lib/puppet/gettext/config.rb

```diff
@@ -75,6 +75,39 @@ def self.create_default_text_domain
     load_translations('puppet', puppet_locale_path, translation_mode(puppet_locale_path))
   end
 
+  # @api private
+  # Switches the active text domain, if the requested domain exists.
+  # @param [String] domain_name the name of the domain to switch to
+  def self.use_text_domain(domain_name)
+    return unless gettext_loaded?
+
+    if FastGettext.translation_repositories[domain_name]
+      FastGettext.text_domain = domain_name
+    end
+  end
+
+  # @api private
+  # Deletes the text domain with the given name
+  # @param [String] domain_name the name of the domain to delete
+  def self.delete_text_domain(domain_name)
+    return unless gettext_loaded?
+
+    FastGettext.translation_repositories.delete(domain_name)
+  end
+
+  # @api private
+  # Deletes all text domains except the default one
+  def self.delete_environment_text_domains
+    return unless gettext_loaded?
+
+    FastGettext.translation_repositories.keys.each do |key|
+      # do not clear default translations
+      next if key == 'puppet'
```

Comment:
Is anyone going to have a `puppet` environment?

Reply:
Might need to pick a better name for the default text domain. I suppose there's always the random chance that someone could use it, but we can make it way less likely to conflict. Maybe `default-text-domain` or something.

Reply:
I think using dashes makes it an [invalid environment name](https://puppet.com/docs/puppet/5.3/lang_reserved.html#environments), which would ensure we don't have collisions with a user-specified environment.

Reply:
Perfect.